Update index.md (#29273)
mdn/content: files/en-us/web/api/document/cookie/index.md

  pair, each preceded by a semicolon separator:

  - `;domain=domain` (e.g.,

>     '`example.com`' or '`subdomain.example.com`'): The host to which the cookie will be sent. If not specified, this defaults to the host portion of the current document location and the cookie is treated as [host-only](https://datatracker.ietf.org/doc/html/rfc6265#section-5.3) (i.e. subdomains are not included). Contrary to earlier specifications, leading dots in domain names are ignored, but browsers may decline to set the cookie containing such dots. If a domain is specified, subdomains are always included.

Reviewer: Thanks very much - it's useful to specify that cookies subdomains are not available in this case. However MDN tries to avoid linking to definitions in the specs except from the specifications section (we try to make things a bit more human readable). In this case we don't need the link to capture the main point 

```suggestion
    '`example.com`' or '`subdomain.example.com`'): The host to which the cookie will be sent. If not specified, this defaults to the host portion of the current document location and the cookie is not available on subdomains.
     If a domain is specified, subdomains are always included.
     Contrary to earlier specifications, leading dots in domain names are ignored, but browsers may decline to set the cookie containing such dots.
```
